### PR TITLE
Fixed OV5640 Imaging Modes

### PR DESCRIPTION
**Pull request by kwagyeman**
* Grayscale/RGB565/Bayer/JPEG works perfectly on all reses.
* Increased maximum JPEG image transfer size.
* Improved line buffer transfer rate.

### CODE REVIEW

File: src/omv/sensor.c [hunk 1]
```diff
@@ -525,6 +525,8 @@ int sensor_set_pixformat(pixformat_t pixformat)
         return -1;
     }
 
+    systick_sleep(100); // wait for the camera to settle
```
Comment (iabdalkader) [inline src/omv/sensor.c:528]: This is usually followed by set_framesize (or the other way around), you'll be adding 200ms delay, and then skip frames is called anyway (or at least 100ms will pass by the time snapshot is called). Is it necessary ?

File: src/omv/sensor.c [hunk 2]
```diff
@@ -888,10 +892,57 @@ static void sensor_check_buffsize()
     }
 }
 
+// ARM Cortex-M4/M7 Processors can access memory using unaligned 32-bit reads/writes.
```
Comment (iabdalkader) [inline src/omv/sensor.c:895]: It does, but it's actually slower.

File: src/omv/sensor.c (hunk 1 above)
Comment (kwagyeman) [reply src/omv/sensor.c:528]: It just hides a corrupt frame that appears when you change modes. I didn't want to put in within the driver as that can be changed via set frame size.

File: src/omv/sensor.c (hunk 2 above)
Comment (kwagyeman) [reply src/omv/sensor.c:895]: Yes, I found an algorithm that does 4 byte aligned writes and does barrel shifting to align things also. Maybe I'll have Larry implement that faster. This was less code and I didn't want to bring in that giant algorithm.

**Pull request merged**